components/kyma-environment-broker/internal/process/upgrade_kyma/evaluation_manager.go

```diff
@@ -0,0 +1,70 @@
+package upgrade_kyma
+
+import (
+	"time"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/avs"
+	"github.com/sirupsen/logrus"
+)
+
+type EvaluationManager struct {
+	avsConfig         avs.Config
+	delegator         *avs.Delegator
+	internalAssistant *avs.InternalEvalAssistant
+	externalAssistant *avs.ExternalEvalAssistant
+}
+
+func NewEvaluationManager(delegator *avs.Delegator, config avs.Config) *EvaluationManager {
+	return &EvaluationManager{
+		delegator:         delegator,
+		avsConfig:         config,
+		internalAssistant: avs.NewInternalEvalAssistant(config),
+		externalAssistant: avs.NewExternalEvalAssistant(config),
+	}
+}
+
+// SetStatus updates evaluation monitors (internal and external) status.
+// Note that this operation should be called twice (reason behind the zero delay)
+// to configure both monitors.
+func (em *EvaluationManager) SetStatus(status string, operation internal.UpgradeKymaOperation, logger logrus.FieldLogger) (internal.UpgradeKymaOperation, time.Duration, error) {
+	avsCfg := operation.Avs
+
+	// do internal monitor status update
+	if em.internalAssistant.IsInMaintenance(avsCfg) {
```

Comment:
Erroneous check for maintenance status, call from `SetMaintenanceStatus` will not set the maintenance status because these conditions skip processing.

Reply:
Good observation!